simulation/sub8_gazebo/nodes/gazebo_controller.py

```diff
@@ -17,33 +19,42 @@ class GazeboInterface(object):
 
     def __init__(self, target='sub8::base_link'):
         self.target = target
-        # rospy.wait_for_service('/gazebo/apply_body_wrench')
-        # self.wrench_srv = rospy.ServiceProxy('/gazebo/apply_body_wrench', ApplyBodyWrench)
+        rospy.wait_for_service('/gazebo/apply_body_wrench')
+        self.wrench_srv = rospy.ServiceProxy(
+            '/gazebo/apply_body_wrench', ApplyBodyWrench)
         # For now, let's skip the wrench
-        # self.wrench_sub = rospy.Subscriber('wrench', WrenchStamped, self.wrench_cb)
+        self.wrench_sub = rospy.Subscriber(
+            'wrench', WrenchStamped, self.wrench_cb)
 
         self.last_odom = None
         self.position_offset = None
-        self.state_sub = rospy.Subscriber('/gazebo/link_states', LinkStates, self.state_cb)
-        self.state_set_pub = rospy.Publisher('/gazebo/set_model_state', ModelState, queue_size=1)
+        self.state_sub = rospy.Subscriber(
+            '/gazebo/link_states', LinkStates, self.state_cb)
+        self.state_set_pub = rospy.Publisher(
+            '/gazebo/set_model_state', ModelState, queue_size=1)
 
-        self.raw_dvl_sub = rospy.Subscriber('dvl/range_raw', Range, self.publish_height)
+        self.raw_dvl_sub = rospy.Subscriber(
+            'dvl/range_raw', Range, self.publish_height)
         self.dvl_pub = rospy.Publisher('dvl/range', RangeStamped, queue_size=1)
 
-        self.raw_blueview_sub = rospy.Subscriber('/blueview_driver/gazebo', LaserScan, self.publish_blueview_ping)
-        self.blueview_ranges_pub = rospy.Publisher('/blueview_driver/ranges', BlueViewPing, queue_size=1)
+        self.raw_blueview_sub = rospy.Subscriber(
+            '/blueview_driver/gazebo', LaserScan, self.publish_blueview_ping)
+        self.blueview_ranges_pub = rospy.Publisher(
+            '/blueview_driver/ranges', BlueViewPing, queue_size=1)
 
-        self.reset_srv = rospy.Service('gazebo/reset_gazebo', ResetGazebo, self.reset)
+        self.reset_srv = rospy.Service(
+            'gazebo/reset_gazebo', ResetGazebo, self.reset)
         self.state_pub = rospy.Publisher('model_odom', Odometry, queue_size=1)
-        self.world_state_pub = rospy.Publisher('world_odom', Odometry, queue_size=1)
+        self.world_state_pub = rospy.Publisher(
+            'world_odom', Odometry, queue_size=1)
 
         self.odom_freq = 0.03
         rospy.Timer(rospy.Duration(self.odom_freq), self.publish_odom)
 
     def send_wrench(self, wrench):
         self.wrench_srv(
             body_name=self.target,
-            reference_frame=self.target,
+            reference_frame='world',
             wrench=wrench,
             duration=rospy.Duration(0.001)
```

Comment:
Didn't this need to be increased?

Reply:
I changed this back and forth and it looked like it was working with this number. It may be more realistic to increase this number.